test-it/karma/publicapi/Double.Test.js

```diff
@@ -138,45 +132,26 @@ describe('The Vireo EggShell Double api can', function () {
-            testWriteDoubleCoerced('uInt32MinValue', 0, -8123456789, expectedCoercedValue(Uint32Array, -8123456789));
-
-            expect(expectedCoercedValue(Int8Array, 812)).toBe(44);
-            expect(expectedCoercedValue(Int8Array, -812)).toBe(-44);
-            expect(expectedCoercedValue(Int16Array, 81234)).toBe(15698);
-            expect(expectedCoercedValue(Int16Array, -81234)).toBe(-15698);
-            expect(expectedCoercedValue(Int32Array, 8123456789)).toBe(-466477803);
-            expect(expectedCoercedValue(Int32Array, -8123456789)).toBe(466477803);
-            expect(expectedCoercedValue(Uint8Array, 812)).toBe(44);
-            expect(expectedCoercedValue(Uint8Array, -812)).toBe(212);
-            expect(expectedCoercedValue(Uint16Array, 81234)).toBe(15698);
-            expect(expectedCoercedValue(Uint16Array, -81234)).toBe(49838);
-            expect(expectedCoercedValue(Uint32Array, 8123456789)).toBe(3828489493);
-            expect(expectedCoercedValue(Uint32Array, -8123456789)).toBe(466477803);
+            testWriteDoubleCoerced('int8MinValue', -128, 812, 127);
+            testWriteDoubleCoerced('int8MinValue', -128, -812, -128);
+            testWriteDoubleCoerced('int16MinValue', -32768, 81234, 32767);
+            testWriteDoubleCoerced('int16MinValue', -32768, -81234, -32768);
+            testWriteDoubleCoerced('int32MinValue', -2147483648, 8123456789, 2147483647);
+            testWriteDoubleCoerced('int32MinValue', -2147483648, -8123456789, -2147483648);
+            testWriteDoubleCoerced('uInt8MinValue', 0, 812, 255);
+            testWriteDoubleCoerced('uInt8MinValue', 0, -812, 0);
+            testWriteDoubleCoerced('uInt16MinValue', 0, 81234, 65535);
+            testWriteDoubleCoerced('uInt16MinValue', 0, -81234, 0);
+            testWriteDoubleCoerced('uInt32MinValue', 0, 8123456789, 4294967295);
+            testWriteDoubleCoerced('uInt32MinValue', 0, -8123456789, 0);
         });
 
         it('to coerce iee754 special values to memory', function () {
-            testWriteDoubleCoerced('int8MinValue', -128, NaN, expectedCoercedValue(Int8Array, NaN));
-            testWriteDoubleCoerced('int8MinValue', -128, Infinity, expectedCoercedValue(Int8Array, Infinity));
-            testWriteDoubleCoerced('int8MinValue', -128, -Infinity, expectedCoercedValue(Int8Array, -Infinity));
-            testWriteDoubleCoerced('int8MinValue', -128, -0, expectedCoercedValue(Int8Array, -0));
-            testWriteDoubleCoerced('int8MinValue', -128, 0, expectedCoercedValue(Int8Array, 0));
-
-            expect(expectedCoercedValue(Int8Array, NaN)).toBe(0);
-            expect(expectedCoercedValue(Int8Array, Infinity)).toBe(0);
-            expect(expectedCoercedValue(Int8Array, -Infinity)).toBe(0);
-            expect(expectedCoercedValue(Int8Array, -0)).toBe(0);
-            expect(expectedCoercedValue(Int8Array, 0)).toBe(0);
+            testWriteDoubleCoerced('int8MinValue', -128, NaN, 127);
```

Comment:
@sanmut does this seem right to you? Double NaN [goes to IntTypeMax](https://github.com/ni/VireoSDK/blob/c8eb11d197b09f7ed2100bdb01cd48cd8f2fa003/source/include/TypeAndDataManager.h#L370)? Eventually the saturating WASM float to int will treat [NaN as zero](https://github.com/WebAssembly/spec/blob/master/proposals/nontrapping-float-to-int-conversion/Overview.md#design). We should align with LabVIEW so if LabVIEW is taking NaN to IntMax during float to int that's what we should replicate (wonder if it's compiler specific / architecture dependent).

Reply:
@sanmut friendly ping. Also maybe @spathiwa has some insight if this behavior is right for NaN handling in float-to-int conversions for LabVIEW?

Reply:
@rajsite 
That's correct; NaN conversion to integer in LabVIEW saturates to maxInt, not zero.
In the CG runtime we don't rely on the compiler to do this because it would be architecture-dependent; we explicitly check for NaN/unordered.

Man, I kind of miss working on Vireo.  (I wonder if the Mac build still works.)

Reply:
> That's correct; NaN conversion to integer in LabVIEW saturates to maxInt, not zero.

Thanks for the confirmation! That resolves my question but @sanmut feel free to unresolve if you have any follow-up.

> Man, I kind of miss working on Vireo. (I wonder if the Mac build still works.)

Might be chances coming up as debugging is figured out :)

Reply:
Been busy with bugs. Craig beat me to it. Thanks Craig.
The only other thing I would recommend is to add some test cases such as conversion of different normal floating point values such as "-3.5, -2.1, 2.1, 3.5" to make sure the rounding behavior is FE_TONEAREST and ensure the default conversion behavior does not get affected with compiler upgrades. This can be added as a GVI test in ASW side too. It is not necessary to do this as part of this PR and not high priority.

Reply:
@sanmut that's interesting, hadn't thought about the rounding modes that are enabled. Looks like [WebAssembly currently only supports nearest](https://webassembly.github.io/spec/core/exec/numerics.html?highlight=rounding#rounding) but there is an interesting recent thread about [supporting more modes](https://github.com/WebAssembly/design/issues/1384#issuecomment-857373480).

@Devika-Tantry mentioning so that we capture the work to add tests verifying rounding modes. (mentioning just-in-case as the thread is marked resolved).

Reply:
Done. Created the [issue ](https://github.com/ni/VireoSDK/issues/684)to add more test cases.